### DISCUSSION TITLE
How to invert colour of figures in bulk?

### DISCUSSION THREAD

**Discussion by Python-Zhou** [ImageMagick/ImageMagick#2124] [Help] [answered]
I would like to invert the colour of numerous figures in a fold in Linux, the code is as follows:

`mogrify /home/yu/Desktop/1 -negate /home/yu/Desktop/2`

The origin file is in path 1, and the target fold is path 2.

However, it's wrong. It points out that "mogrify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/562."

Looking forward to your help.

**Comment by fmw42** [comment, marked as answer]
Be sure /home/yu/Desktop/2 already exists. The try

`mogrify  -path /home/yu/Desktop/2 -negate /home/yu/Desktop/1/*`

See https://imagemagick.org/Usage/basics/#mogrify

**Reply by Python-Zhou** [reply]
Thank you very much!

**Comment by fmw42** [comment]
If you are on DOS Windows, it use \ not /.  If on Cygwin Windows, then use /.  If using IM 7, then use magick mogrify and if you do not want to negate the alpha channel, then preface -negate with -channel rgb.  IM 7 negates all channels (RGBA) whereas IM 6 ignored the alpha and only negated the RGB channels.
